### ISSUE TITLE
Better output for resolvers plugins when used with mappers

### ISSUE THREAD

**Issue by dotansimha** [dotansimha/graphql-code-generator#1595]
Given the following schema:

```graphql
type A {
   b: B
}

type B {
   id: ID
}
```

If the configuration of `mappers` include the following:
```
mappers:
   B: MyB
```

Then anywhere in the resolvers, `B` should be replace with `MyB`, but `A` should be `Omit<A, 'b'> & { b: MyB } `, in order to replace the type of `B` with the actual type.
If `A` also has a mapper we should ignore this behaviour and just use the mapped `A`. 

> In flow, we need to use `$Diff`.

**Comment by dotansimha** [comment]
WIP here: https://github.com/dotansimha/graphql-code-generator/pull/1593

**Comment by dotansimha** [comment]
Implemented in 1.0.7 :)
